btas/generic/cp.h

```diff
@@ -347,41 +351,41 @@ namespace btas{
     Tensor reconstruct() {
       if (A.empty()) BTAS_EXCEPTION(
               "Factor matrices have not been computed. You must first calculate CP decomposition.");
-      std::vector<int> dims;
-      for(auto i = 0; i < ndim; ++i){
+      std::vector<size_t> dims;
+      for (size_t i = 0; i < ndim; ++i) {
         dims.push_back(i);
       }
       return btas::reconstruct(A, dims);
     }
 
     // For debug purposes
     void print(Tensor& tensor){
-      if(tensor.rank() == 2){
-        int row = tensor.extent(0), col = tensor.extent(1);
-        for(int i = 0; i < row; ++i){
-          const auto * tensor_ptr = tensor.data() + i * col;
-          for(int j = 0; j < col; ++j){
+      if(tensor.rank() == 2) {
+        ind_t row = tensor.extent(0), col = tensor.extent(1);
```

Comment:
the cleanest way to write this is a la lines "457-459" i.e. use ord_t for `i_times_col`, everything else is `ind_t` ... that way the index types are clear ... seperation of ind_t and ord_t is useful e.g. if the user decides to use `short` for mode indices but `int64_t` for ordinals